optimizely/helpers/audience.py

```diff
@@ -11,26 +11,41 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import json
+
 from . import condition as condition_helper
 from . import condition_tree_evaluator
+from .enums import AudienceEvaluationLogs as logs
 
 
-def is_user_in_experiment(config, experiment, attributes):
+def is_user_in_experiment(config, experiment, attributes, logger):
   """ Determine for given experiment if user satisfies the audiences for the experiment.
 
   Args:
     config: project_config.ProjectConfig object representing the project.
     experiment: Object representing the experiment.
     attributes: Dict representing user attributes which will be used in determining
                 if the audience conditions are met. If not provided, default to an empty dict.
+    logger: Provides a logger to send log messages to.
 
   Returns:
     Boolean representing if user satisfies audience conditions for any of the audiences or not.
   """
 
-  # Return True in case there are no audiences
   audience_conditions = experiment.getAudienceConditionsOrIds()
+
+  logger.debug(logs.EVALUATING_AUDIENCES_COMBINED.format(
+    experiment.key,
+    json.dumps(audience_conditions)
```

Comment:
@aliabbasrizvi yeah, same message in other SDKs... Part of my reasoning was that log messages might be recorded and used for debugging after the fact, by which time the datafile may have changed. Currently it's very difficult (and it may _always_ be unintuitive) for customers to look up the datafile that was contemporaneous with a particular log message.